### ISSUE TITLE
NUnit Test Adaptor 3.16.0 never displays results of test under Visual Studio 2015

### ISSUE THREAD

**Issue by ryanewtaylor** [nunit/nunit3-vs-adapter#692]
# Scene:

1. Window 10
2. Microsoft Visual Studio Professional 2015 (v14,.0.25431.01 Update 3)
3. NUnit 3.12.0
4. NUnit Test Adaptor 3.16.0
5. .NET Framework 4.5.2, x86

# Repro:

* New class library project with one test fixture and test

```csharp
using System;
using NUnit.Framework;

namespace UnitTests
{
    [TestFixture]
    public class UnitTest1
    {
        [Test]
        public void TestMethod1()
        {
            Assert.AreEqual("hi", "bye");
        }
    }
}
```

# Problem:

The results of the test are never fed back into the adapter. I always receive an error in the output window such as 

> ------ Run test started ------
NUnit Adapter 3.16.0.0: Test execution started
Running selected tests in C:\Dev\temp\TestNUnit\UnitTests\bin\x86\Debug\UnitTests.dll
   NUnit3TestExecutor converted 1 of 1 NUnit test cases
NUnit Adapter 3.16.0.0: Test execution complete
Test adapter sent back a result for an unknown test case. Ignoring result for 'TestMethod1'.
========== Run test finished: 0 run (0:00:05.0390741) ==========

The test adapter can discover tests. I can also debug tests.

# Temporary Solution:

Revert back to NUnit Test Adaptor 3.15.1.0 which does not seem to exhibit this issue.

**Comment by lorenh** [comment]
I am also seeing this issue in Visual Studio Enterprise 2019, Version 16.4.2.  I had to revert to 3.15.1.0 of the test adapter as well.  The tests seem to run, but no Green check marks appeared in the Test Explorer. They all just stayed gray

**Comment by OsirisTerje** [comment]
@lorenh I assume you have some parameterized tests in there?  Some based on TestCaseSource too?

**Comment by lorenh** [comment]
Yes, we do have parameterized tests (using TestCase attribute), but none using TestCaseSource at the moment.

```        [Test]
        [TestCase(-180.001, 0, 1, 1)]
        [TestCase(0, -90.001, 1, 1)]
        [TestCase(0, 0, 180.001, 1)]
        [TestCase(0, 0, 1, 90.001)]
        public void BoxGeoFilter_Create_ShouldFailIfOutOfRange(double west, double south, double east, double north)
        {
        : 
        }
```

**Comment by OsirisTerje** [comment]
That should work.....
If I copy what you have above into a seperate test class, they work fine.  So there must be one of the other issues going on here.  Have you seen the other latest issues related to the same symptoms ?

**Comment by lorenh** [comment]
I have not been monitoring any other of the latest issues with these symptoms. Am I safe to assume that "you're working on it", or do you need any more information from me?

**Comment by OsirisTerje** [comment]
I'm absolutely working on them :-)

**Comment by sanhin** [comment]
@ryanewtaylor - In the temporary solution you mentioned about reverting back to NUnit Test Adaptor 3.15.1.0 . I tried few times to uninstall the new version and install 3.15.1.0 but it auto updates it back to the newer version (3.16.0).

Note: I did uncheck  the 'Automatically update this extension' checkbox

![image](https://user-images.githubusercontent.com/56365112/72177908-953aef80-3396-11ea-8d75-b171fa13b7fd.png)

**Comment by OsirisTerje** [comment]
@sanhin If you have the new SDK csproj format or are using PackageReferences , just edit the csproj file directly.
If you're using the old csproj format with package config files, then check the output window and see what messages are there, possibly some dependencies are wrong, blocking your downgrade. E.g. you have dependencies between test projects or something.

**Comment by ryanewtaylor** [comment]
@sanhin This worked for me, YMMV.

I uninstalled the NUnit Test Adaptor and restarted Visual Studio to ensure it was removed. Soemtimes, it remained installed 🤷‍♂.  I repeated the uninstall and restart a couple times until it no longer showed in Extensions and Updates.

Then, with Visual Studio closed I installed the 3.15.1.0 version through the [releases](https://github.com/nunit/nunit3-vs-adapter/releases/download/V3.15.1/NUnit3TestAdapter-3.15.1.vsix). Finally, I unchecked the "Automatically update this extension" and I have remained at v3.15.1.0 for a few days.

**Comment by OsirisTerje** [comment]
This issue is resolved by Version 3.16.1 which is released now on [nuget.org](https://www.nuget.org/packages/NUnit3TestAdapter/3.16.1) and VSIX on [Visual Studio marketplace](https://marketplace.visualstudio.com/items?itemName=NUnitDevelopers.NUnit3TestAdapter).  See [all issues resolved](https://github.com/nunit/nunit3-vs-adapter/issues?q=is%3Aissue+milestone%3A3.16.1+is%3Aclosed) and  [Release notes](https://github.com/nunit/docs/wiki/Adapter-Release-Notes) for details.
